### PR TITLE
[3,4,5] Added movie model, controller and views + categories

### PR DESCRIPTION
**Pull request by justynapaldyna**

### CODE REVIEW

File: app/controllers/categories_controller.rb [hunk 1]
```diff
@@ -0,0 +1,65 @@
...
+    @category.destroy
+
+    respond_to do |format|
+      format.html { redirect_to categories_url, notice: "Category was successfully destroyed." }
+    end
+  end
+
+  private
+    # Use callbacks to share common setup or constraints between actions.
+    def set_category
```
Comment (paleksandrowicz) [inline app/controllers/categories_controller.rb]: indentation

File: app/controllers/movies_controller.rb [hunk 2]
```diff
@@ -0,0 +1,70 @@
+class MoviesController < ApplicationController
+  before_action :set_movie, only: %i[ show edit update destroy ]
+  before_action :authenticate_user!, except: [:show, :index]
+  before_action :set_categories
```
Comment (paleksandrowicz) [inline app/controllers/movies_controller.rb]: do we want to set categories before everything?

File: app/controllers/movies_controller.rb [hunk 3]
```diff
@@ -0,0 +1,70 @@
...
+    @movies = Movie.all
+  end
+
+  # GET /movies/1 or /movies/1.json
+  def show
+  end
+
+  # GET /movies/new
+  def new
+    @movie = current_user.movies.build
```
Comment (paleksandrowicz) [inline app/controllers/movies_controller.rb]: Why not Movie.new?

File: app/controllers/movies_controller.rb [hunk 4]
```diff
@@ -0,0 +1,70 @@
...
+    @movie.destroy
+
+    respond_to do |format|
+      format.html { redirect_to movies_url, notice: "Movie was successfully destroyed." }
+    end
+  end
+
+  private
+    # Use callbacks to share common setup or constraints between actions.
+    def set_movie
```
Comment (paleksandrowicz) [inline app/controllers/movies_controller.rb]: indentation

File: app/models/movie.rb [hunk 5]
```diff
@@ -0,0 +1,14 @@
+class Movie < ApplicationRecord
+    belongs_to :user
+    belongs_to :category
+    has_one_attached :image
+    validates :category_id, presence: true
+    validate :correct_image_file_type
+    private
+
+    def correct_image_file_type
+        if image.attached? && ['image/png', 'image/jpg', 'image/jpeg'].exclude?(image.content_type)
```
Comment (paleksandrowicz) [inline app/models/movie.rb:13]: indenation

File: app/views/categories/_category.json.jbuilder [hunk 6]
```diff
@@ -0,0 +1,2 @@
+json.extract! category, :id, :name, :created_at, :updated_at
```
Comment (paleksandrowicz) [inline app/views/categories/_category.json.jbuilder]: where do you use jbuilder files?

Comment (justynapaldyna) [reply app/views/categories/_category.json.jbuilder]: from  scaffold - will delete

File: app/controllers/movies_controller.rb (hunk 3 above)
Comment (justynapaldyna) [reply app/controllers/movies_controller.rb]: I wanted to connect the movie to an user that added it, to later check if he is an admin for example

**Comment by justynapaldyna** [comment]
I still have to corret the indentation you wrote about, but i tried to write few tests -  user model test  and categories_controller test are still to be written, i hope i will have time to finish them on weekend :) tests still have some failures, but i wanted to update here what i am working at right now

File: spec/controllers/movies_controller_spec.rb [hunk 7]
```diff
@@ -0,0 +1,119 @@
+require 'rails_helper'
+
+RSpec.describe MoviesController, type: :controller do
+
+    describe 'GET #index' do
+        subject { get :index }
```
Comment (paleksandrowicz) [inline spec/controllers/movies_controller_spec.rb]: indentations :)

**Pull request merged**